internal/backends/nodejs/nodejs.go

```diff
@@ -309,6 +309,67 @@ var NodejsYarnBackend = api.LanguageBackend{
 	Guess:        nodejsGuess,
 }
 
+var NodejsPNPMBackend = api.LanguageBackend{
+	Name:             "nodejs-pnpm",
+	Specfile:         "package.json",
+	Lockfile:         "pnpm-lock.yaml",
+	FilenamePatterns: nodejsPatterns,
+	Quirks: api.QuirksAddRemoveAlsoLocks |
+		api.QuirksAddRemoveAlsoInstalls |
+		api.QuirksLockAlsoInstalls,
+	GetPackageDir: func() string {
+		return "node_modules"
+	},
+	Search: nodejsSearch,
+	Info:   nodejsInfo,
+	Add: func(pkgs map[api.PkgName]api.PkgSpec, projectName string) {
+		if !util.Exists("package.json") {
+			util.RunCmd([]string{"npm", "init", "-y"})
```

Comment:
Why not `pnpm init`?

Reply:
yeah, this should be `pnpm init` - we should only use the tools that are being used in the environment. shouldn't rely on `npm` being available if we're only promised `pnpm`

Reply:
I did test using `pnpm init` and found that it doesn't have an option to generate with the defaults, but instead throws the user into a dialog, where they have to confirm options. Perhaps I made a typo and accidentally used `npm` when testing that, because that doesn't seem to happen if I test it again.

Reply:
yeah, `pnpm` works for me without any interaction.